code/workhorse/OOPGP.R

```diff
@@ -0,0 +1,118 @@
+# Define a class named Simulator.
+# The main method used in main.R is the simulate method. Other methods of this class and modules in other files are helper functions 
+# for the simulate method.
+
```

Comment:
Using multiple blank lines is goes against the convention (here and elsewhere).  Use only a single blank line when needed.

Reply:
Replace double lines with a single line. In pep8, it is recommended to surround top-level function and class definitions with two blank lines.